tests/UnitTests/CoreGraphics.drawing/CGContextDrawingTests.cpp

```diff
@@ -44,6 +44,128 @@ DISABLED_DRAW_TEST_F(CGContext, DrawIntoRect, UIKitMimicTest) {
 }
 #endif
 
+DRAW_TEST_F(CGContext, LinearGradient, UIKitMimicTest) {
+    CGContextRef context = GetDrawingContext();
+
+    CGFloat locations[2] = { 1.0, 0.0 };
+    CGFloat components[8] = { 0.0, 0.5, 0.0, 1.0, 1.0, 1.0, 0.8, 1.0 };
+    woc::unique_cf<CGColorSpaceRef> rgbColorSpace(CGColorSpaceCreateDeviceRGB());
+    CGGradientRef gradient = CGGradientCreateWithColorComponents(rgbColorSpace.get(), components, locations, 2);
+
+    CGPoint startPoint = CGPointMake(0, 0);
+    CGPoint endPoint = CGPointMake(512, 1024);
+
+    CGContextDrawLinearGradient(context, gradient, startPoint, endPoint, kCGGradientDrawsBeforeStartLocation);
+}
+
+DRAW_TEST_F(CGContext, LinearGradientInvalidCount, UIKitMimicTest) {
+    CGContextRef context = GetDrawingContext();
+    woc::unique_cf<CGColorSpaceRef> rgbColorSpace(CGColorSpaceCreateDeviceRGB());
+
+    CGFloat locations[] = { 0.0 };
+
+    CGFloat components[] = {
+        0.85, 0, 0, 1.0,
+    };
+
+    CGColorSpaceRef colorspace = CGColorSpaceCreateDeviceRGB();
```

Comment:
technically we are only going to be able to fully modularize 2-3 lines and we will have multitude of variables being passed in, which ends up being just like the DrawLinearGradient.


---
In reply to: [93269632](https://github.com/Microsoft/WinObjC/pull/1535#discussion_r93269632) [](ancestors = 93269632)